Feature/qna2

Tekkon opened the pull request:
Реализация историй "Пользователь может создавать вопрос" и "Пользователь может создавать ответ на вопрос"

Commit:
Контроллер QuestionControllers и тесты для него

Commit:
Вьюхи для модели Question

Commit:
Добавил вложенный ресурс answers

Commit:
Создал AnswersController и тесты для него

vkurennov commented on app/controllers/answers_controller.rb:
```diff
@@ -0,0 +1,27 @@
+  before_action :find_question, only: [:new, :create]
+
+  def new
+  	@answer = @question.answers.build
```
Что с отступами? Отступы должны быть в 2 пробела, не больше

vkurennov commented on app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,50 @@
+  def create
+  	@question = Question.new(question_params)
+
+  	if @question.save
```
Следи за отступами, поставь какой-нибудь редактор, который умеет автоматически выравнивать код и сделай автоформатирование для существующего кода.Такое, например, умеет RubyMine

vkurennov commented on spec/controllers/answers_controller_spec.rb:
```diff
@@ -0,0 +1,41 @@
+
+  	context 'with invalid attributes' do
+  	  it 'does not save the question' do
+  	  	expect { post :create, params: { question_id: question, answer: attributes_for(:invalid_answer) } }.to_not change(question.answers, :count)
```
Здесь должен быть просто Answer в change, т.к. важно проверить, что вообще никакой ответ не был создан, вне зависимости от его связей с вопросом

Commit:
1. Добавил idea в gitignore
2. Исправил отступы
3. Внес исправление в answers_controller_spec: в тесте на создание Answer с невалидными параметрами проверяем, что кол-во объектов не изменилось у всех Answer
